### PR TITLE
[chore] Remove go-playground/validator

### PR DESCRIPTION
**Pull request by daenney**
# Description

It turns out we're not actually using the validator code. This is a remnant from when we intended to use it, but the presence of it and its struct tags creates the illusion we're validating a lot of things we're not. It resulted in some confusion when we were trying to figure out language validation.

Remove all this code, so that only the validation functions from the validate package we actually use remain. I'm not touching the struct tags in the migrations in order to avoid things potentially thinking migrations need to be re-run.

## Checklist

Please put an x inside each checkbox to indicate that you've read and followed it: `[ ]` -> `[x]`

If this is a documentation change, only the first checkbox must be filled (you can delete the others if you want).

- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [ ] I/we have performed a self-review of added code.
- [ ] I/we have written code that is legible and maintainable by others.
- [ ] I/we have commented the added code, particularly in hard-to-understand areas.
- [ ] I/we have made any necessary changes to documentation.
- [ ] I/we have added tests that cover new code.
- [x] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

File: internal/api/model/domain.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ import "mime/multipart"
 type Domain struct {
 	// The hostname of the domain.
 	// example: example.org
-	Domain string `form:"domain" json:"domain" validate:"required"`
+	Domain string `form:"domain" json:"domain"`
```
Comment (tsmethurst) [inline internal/api/model/domain.go:28]: This tag should stay, since it pertains to gin form parsing validation, not struct validation

**Comment by tsmethurst** [comment]
Looks good, with that one comment aside, thanks for doing this :)

File: internal/api/model/domain.go (hunk 1 above)
Comment (daenney) [reply internal/api/model/domain.go:28]: Ah, right. How nice of people to use the same struct tags :sweat_smile:

Comment (tsmethurst) [reply internal/api/model/domain.go]: Right!

**Pull request merged**